pkg/currency/updater.go

```diff
@@ -59,16 +71,27 @@ func NewUpdater(ctx context.Context, config cfg.Config, logger log.Logger) (Upda
 		return nil, fmt.Errorf("failed to unmarshal currency updater settings: %w", err)
 	}
 
-	return NewUpdaterWithInterfaces(logger, store, httpClient, clock.Provider, settings), nil
+	providersConfig := make(ProvidersConfig)
+	if err := config.UnmarshalKey("currency_service.providers", &providersConfig); err != nil {
```

Comment:
This doesn't work if I configure my provider completely via env variables. You have to loop over all registered providers and try to get their config manually